utils/build_profiler_report.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python3
+import sys
+
+
+def LoadLogFile(file) -> dict:
+    """Loads Targets and their durations from ninja logfile `file` and returns them in a dict"""
+
+    with open(file, 'r') as fp:
+        lines = fp.read().splitlines()
+
+    # {Target: duration} dict
+    TargetDurationDict = {}
+    for line in lines:
+        line = line.split()
+
+        # ToDo - create better comment detection
+        if line[0][0] != '#':
+
+            # calculate target compilation duration and add it to dict
+            duration = int(line[1]) - int(line[0])
+            TargetDurationDict[line[3]] = duration
+
+    return TargetDurationDict
+
+
+def FormatTime(t):
+    """Converts a time into a human-readable format"""
+
+    t /= 1000
+    if t < 60:
+        return '%.1fs' % t
+    if t < 60 * 60:
+        return '%dm%.1fs' % (t / 60, t % 60)
+    return '%dh%dm%.1fs' % (t / (60 * 60), t % (60 * 60) / 60, t % 60)
+
+
+def GenerateWebTreeMap(CurrentDict: dict, BaselineDict: dict) -> None:
+    """Create file for webtreemap to generate an HTML from; if target is new, append _NEW"""
+    with open(sys.argv[1] + ".webtreemap", 'w') as fp:
+        for target in CurrentDict.keys():
+            new_tag = ''
+            if BaselineDict and target not in BaselineDict.keys():
+                new_tag = '_NEW'
+
+            fp.write(str(CurrentDict[target]) + ' ' + target
+                     + '_' + FormatTime(CurrentDict[target]) + new_tag + '\n')
+
+
+def GetTotalTime(TargetDurationDict: dict) -> int:
+    """Return sum of durations for all targets in dict"""
+    total_time = 0
+    for target in TargetDurationDict.keys():
+        total_time += TargetDurationDict[target]
+    return total_time
+
+
+def GetTotalTimeIntersect(TargetDurationDict_A: dict, TargetDurationDict_B: dict) -> int:
+    """Return sum of durations for targets in A that are also in B"""
+    total_time = 0
+    for target in TargetDurationDict_A.keys():
+        if target in TargetDurationDict_B.keys():
+            total_time += TargetDurationDict_A[target]
+    return total_time
+
+
+def PrintReport(CurrentDict: dict, BaselineDict: dict = None) -> None:
+    """Print report with results of profiling to stdout"""
+
+    # If the targets have changed between baseline and current, we are using total_time_intersect
+    # to calculate delta (ratio of durations of targets) instead of total_time
+    if BaselineDict and BaselineDict.keys() != CurrentDict.keys():
+        print("Warning: the targets in the current logfile differ from those in the baseline\
+             logfile; therefore the Δ delta is calculated without taking the added/removed\
+                  targets into consideration. If the added/removed targets modify the behavior of\
+                      targets in both logfiles, the Δ delta may not make sense.\n-----")
```

Comment:
Coding style! You don't have to use `\` if you're inside parenthesis, you can put `"` at the end of the line, and then `"` at the beginning of the new line (after leading spaces), and the strings will get concatenated.

Reply:
I would suggest to create a variable holding the message first, and then just print the variable.

Also I would use D or delta instead of the greek capital letter, that way nobody will scratch their heads because of mysterious triangles, and you will avoid potential encoding issues.